syntax = "proto3";

package towercontrol;

// The greeting service definition.
service Tower {

  rpc SayLandingTrack (LandingTrackRequest) returns (LandingTrackReply){}

  rpc SayDepartureTrack (DepartureTrackRequest) returns (DepartureTrackReply){}

  rpc SayFlights (FlightsRequest) returns (stream FlightsResponse){}
}


message AltitudeRequest {
  int32 altitude = 1;
}

// The response message containing the greetings
message AltitudeReply {
  int32 message = 1;
}

message LandingTrackRequest {
  int32 track = 1;
  int32 altitude = 2;
  string flightnumber = 3;
  string destiny = 4;
}

// The response message containing the greetings
message LandingTrackReply{
  int32 message = 1;
  int32 pos = 2;
  int32 altitude = 3;
}

message DepartureTrackRequest {
  string flightnumber = 1;
  string destiny = 2;
}

// The response message containing the greetings
message DepartureTrackReply {
  int32 track = 1;
  int32 height = 2;
  string ip = 3;
  string port = 4;
  int32 pos = 5;
}

message FlightsRequest{
  int32 message = 1;
} 

message FlightsResponse{
  string type = 1;
  string airport = 2;
  string flight = 3;
  string destiny = 4;
  int32 track = 5;
}
